### PR TITLE
Fix bug with Seek() and optimise bounding params.

### PR DESCRIPTION
**Pull request by Gouthamve**
Started out as a minor optimisation but revealed a bug :smile:

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/tsdb/97)

### CODE REVIEW

**Comment by fabxc** [comment]
Can you add a regression test?

File: querier.go [hunk 1]
```diff
@@ -708,9 +706,15 @@ func (it *chunkSeriesIterator) At() (t int64, v float64) {
 func (it *chunkSeriesIterator) Next() bool {
 	for it.cur.Next() {
 		t, _ := it.cur.At()
-		if it.inBounds(t) {
-			return true
+		if t < it.mint {
+			return it.Seek(it.mint)
+		}
```
Comment (fabxc) [inline querier.go]: Was not doing a seek here before potentially a second bug?

Comment (Gouthamve) [inline querier.go]: No it wasn't, it was just inefficient.

**Comment by Gouthamve** [comment]
Added regression test. PTAL.

**Pull request merged**